src/List/organisation-unit-list/OrganisationUnitList.component.js

```diff
@@ -35,13 +35,14 @@ export default class OrganisationUnitList extends React.Component {
                         .filter().on('parent.id').equals(selectedOrganisationUnit.id)
                         .list({ fields: fieldFilteringForQuery });
 
-                    // When a root organisation unit is selected we also add the root organisation unit to the list
-                    // of available organisation units to pick from
-                    if (userOrganisationUnitIds.indexOf(selectedOrganisationUnit.id) >= 0) {
-                        organisationUnitList.add(selectedOrganisationUnit);
-                    }
+                    // DHIS2-2160 Add the selected node to the list to
+                    // avoid having to select the parent node to edit
+                    // the selected node...
+                    let filteredOrgUnits = organisationUnitList.toArray();
+                    filteredOrgUnits.unshift(selectedOrganisationUnit);
+                    let prependedOrgUnitList = ModelCollection.create(d2.models.organisationUnit, filteredOrgUnits);
```

Comment:
I have looked for an alternative way to do this, but also couldn't find anything better. There is the `valuesContainerMap` which has an `[[Entries]]` symbol. This is an array. God knows what happens when you unshift straight into that. The solution you have is correct and reads well. Performance could well be quite poor because you must be iterating through stuff at least twice, but I don't think this is an issue, since we are already working with a response that was filtered on the server.
I say go for it :-)